src/main/scala/ai/datahunters/md/reader/SolrHashReader.scala

```diff
@@ -0,0 +1,81 @@
+package ai.datahunters.md.reader
+
+import java.util
+
+import ai.datahunters.md.config.writer.SolrWriterConfig
+import ai.datahunters.md.util.SolrClientBuilder
+import org.apache.solr.client.solrj.impl.CloudSolrClient
+import org.apache.solr.common.SolrDocument
+import org.apache.solr.common.params.MapSolrParams
+import org.apache.spark.sql.types.{DataTypes, StructField, StructType}
+import org.apache.spark.sql.{DataFrame, Row, SparkSession}
+
+import scala.collection.JavaConversions._
+
+/**
+  * Read  id and hashes to DataFrame mainly for processing reduction
+  *
+  * @param sparkSession
+  * @param config
+  */
+case class SolrHashReader(sparkSession: SparkSession,
+                          config: SolrWriterConfig) extends PipelineSource {
+
+  import ai.datahunters.md.reader.SolrHashReader._
+
+  private val documentToStrings: SolrDocument => Seq[String] =
+    (document: SolrDocument) => HashSeq.map(s => getDocumentFieldValueOrEmptyString(document, s))
+
+
+  override def load(): DataFrame = {
+    val client = SolrClientBuilder().setZKServers(config.zkServers)
```

Comment:
@dawidsau I don't remember where we should add it but probably here - setting kerberos props if exists. Please check in Solr Writer classes.